File: VB/WindowsApplication32/Form1.vb
Imports DevExpress.XtraPivotGrid
Imports System
Imports System.Windows.Forms

Namespace WindowsApplication32
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			' TODO: This line of code loads data into the 'nwindDataSet.CustomerReports' table. You can move, or remove it, as needed.
			Me.customerReportsTableAdapter.Fill(Me.nwindDataSet.CustomerReports)

			InitPivotGrid(pivotGridControl1)
			pivotGridControl1.FieldsCustomization(splitContainer1.Panel1)
		End Sub

		Private Sub InitPivotGrid(ByVal pivot As PivotGridControl)
			pivot.OptionsCustomization.CustomizationFormStyle = DevExpress.XtraPivotGrid.Customization.CustomizationFormStyle.Excel2007
			pivot.RetrieveFields(PivotArea.FilterArea, False)

			pivot.Fields("ProductAmount").Area = PivotArea.DataArea
			pivot.Fields("ProductAmount").Visible = True
			pivot.Fields("CompanyName").Area = PivotArea.RowArea
			pivot.Fields("CompanyName").Visible = True

			DirectCast(pivot.Fields("OrderDate").DataBinding, DataSourceColumnBinding).GroupInterval = PivotGroupInterval.DateYear
		End Sub
	End Class
End Namespace

File: CS/WindowsApplication32/Form1.cs
using DevExpress.XtraPivotGrid;
using System;
using System.Windows.Forms;

namespace WindowsApplication32
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            // TODO: This line of code loads data into the 'nwindDataSet.CustomerReports' table. You can move, or remove it, as needed.
            this.customerReportsTableAdapter.Fill(this.nwindDataSet.CustomerReports);

            InitPivotGrid(pivotGridControl1);            
            pivotGridControl1.FieldsCustomization(splitContainer1.Panel1);
        }

        private void InitPivotGrid(PivotGridControl pivot)
        {
            pivot.OptionsCustomization.CustomizationFormStyle = DevExpress.XtraPivotGrid.Customization.CustomizationFormStyle.Excel2007;
            pivot.RetrieveFields( PivotArea.FilterArea, false);

            pivot.Fields["ProductAmount"].Area = PivotArea.DataArea;
            pivot.Fields["ProductAmount"].Visible = true;
            pivot.Fields["CompanyName"].Area = PivotArea.RowArea;
            pivot.Fields["CompanyName"].Visible = true;

            (pivot.Fields["OrderDate"].DataBinding as DataSourceColumnBinding).GroupInterval = PivotGroupInterval.DateYear;
        }
    }
}